### PR TITLE
Fix rendering the mapped parameters when using `expand_kwargs` method

### LINKED ISSUE #32260: Apparently the Jinja template does not work when using dynamic task mapping with SQLExecuteQueryOperator
### Apache Airflow version

2.6.2

### What happened

We are trying to use dynamic task mapping with SQLExecuteQueryOperator on Trino. Our use case is to expand the sql parameter to the operator by calling some SQL files. 

Without dynamic task mapping it works perfectly, but when used with the dynamic task mapping, it is unable to recognize the Path, and instead tries to execute the path as query. 

I believe it has some relation with the template_searchpath parameter.

### What you think should happen instead

It should have worked similar with or without dynamic task mapping.

### How to reproduce

Deployed the following DAG in Airflow 

```
from airflow.models import DAG
from datetime import datetime, timedelta
from airflow.providers.common.sql.operators.sql import SQLExecuteQueryOperator

DEFAULT_ARGS = {
    'start_date': datetime(2023, 7, 16),
}

with DAG (dag_id= 'trino_dinamic_map',
    template_searchpath = '/opt/airflow',
	description = "Esta é um dag para o projeto exemplo",
    schedule = None,
	default_args = DEFAULT_ARGS,
    catchup=False,
) as dag:


    trino_call = SQLExecuteQueryOperator(
        task_id= 'trino_call',
        conn_id='con_id',
        sql = 'queries/insert_delta_dp_raw_table1.sql',
        handler=list
    )

    trino_insert = SQLExecuteQueryOperator.partial(
        task_id="trino_insert_table",
        conn_id='con_id',
        handler=list
    ).expand_kwargs([{'sql': 'queries/insert_delta_dp_raw_table1.sql'}, {'sql': 'queries/insert_delta_dp_raw_table2.sql'}, {'sql': 'queries/insert_delta_dp_raw_table3.sql'}])


trino_call >> trino_insert
```

In the sql file it can be any query, for the test I used a create table. Queries are located in /opt/airflow/queries

```
CREATE TABLE database_config.data_base_name.TABLE_NAME (
    "JOB_NAME" VARCHAR(60) NOT NULL,
    "JOB_ID" DECIMAL NOT NULL,
    "JOB_STATUS" VARCHAR(10),
    "JOB_STARTED_DATE" VARCHAR(10),
    "JOB_STARTED_TIME" VARCHAR(10),
    "JOB_FINISHED_DATE" VARCHAR(10),
    "JOB_FINISHED_TIME" VARCHSAR(10)
)
```

task_1 (without dynamic task mapping) completes successfully, while task_2(with dynamic task mapping) fails.

Looking at the error logs, there was a failure when executing the query, not recognizing the query content but the path.
Here is the traceback:

trino.exceptions.TrinoUserError: TrinoUserError(type=USER_ERROR, name=SYNTAX_ERROR, message="line 1:1: mismatched input 'queries'. Expecting: 'ALTER', 'ANALYZE', 'CALL', 'COMMENT', 'COMMIT', 'CREATE', 'DEALLOCATE', 'DELETE', 'DENY', 'DESC', 'DESCRIBE', 'DROP', 'EXECUTE', 'EXPLAIN', 'GRANT', 'INSERT', 'MERGE', 'PREPARE', 'REFRESH', 'RESET', 'REVOKE', 'ROLLBACK', 'SET', 'SHOW', 'START', 'TRUNCATE', 'UPDATE', 'USE', <query>", query_id=20230629_114146_04418_qbcnd)

### Operating System

Red Hat Enterprise Linux 8.8

### Versions of Apache Airflow Providers

_No response_

### Deployment

Docker-Compose

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #32260

This PR fixes a bug in the mapped operator when using expand_kwargs method. 

In the `ListOfDictsExpandInput` class, the resolve method doesn't filter out string values from the `resolved_oids` list as we do in the class `DictOfListsExpandInput`. Consequently, these strings are incorrectly regarded as resolved, even though they are not instances from `MappedArgument` or `XComArg`, and their values still match those provided by the user.

This PR removes the literal str values from the list `resolved_oids` to let the BaseOperator class resolve them.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by hussein-awala** [comment]
I have tested this with a testing DAG and confirmed that it works as expected. However, before removing the `WIP` status, it should also be tested with a taskflow DAG to ensure its compatibility and stability.

**Comment by uranusjr** [comment]
Tests disagree 🙂

`_expand_mapped_kwargs` performs some rendering, and the returned OIDs record those already-performed rendering and prevent double-rendering. This broke the mechanism.

**Comment by hussein-awala** [comment]
> _expand_mapped_kwargs performs some rendering, and the returned OIDs record those already-performed rendering and prevent double-rendering. This broke the mechanism.

@uranusjr According to the failed test, we should use the XCom data as it is without resolving it via Jinja. In this case, we cannot consider the reported issue as a bug, because the task result is shared using XCom, and not resolving this result is an expected behavior.

However, supporting passing templates via XCom especially for the mapped operator is a nice feature IMO, for that, I propose adding a new parameter `template_in_template` to the expand method to tell Airflow whether to resolve the resolved parameters or not.

WDYT?

**Comment by uranusjr** [comment]
Isn’t the original issue about passing a template as a string literal? That sounds like a reasonable thing, and doesn’t involve XCom.

**Comment by hussein-awala** [comment]
@uranusjr, I discovered that the issue is just related to the `expand_kwargs` method. In the `ListOfDictsExpandInput` class, the `resolve` method doesn't filter out string values from the `resolved_oids` list as we do in the class `DictOfListsExpandInput`. Consequently, these strings are incorrectly regarded as resolved, even though they are not instances from `MappedArgument` or `XComArg`, and  their values still match those provided by the user.

Could you take another look?

For the new feature "template_in_template" I proposed, I will open a new PR to discuss its idea.

File: airflow/models/expandinput.py [hunk 1]
```diff
@@ -217,6 +217,10 @@ class ListOfDictsExpandInput(NamedTuple):
 
     value: OperatorExpandKwargsArgument
 
+    def _iter_parse_time_resolved_kwargs(self, mapping) -> Iterable[tuple[str, Sized]]:
+        """Generate kwargs with values available on parse-time."""
+        return ((k, v) for k, v in mapping.items() if _is_parse_time_mappable(v))
```
Comment (uranusjr) [inline airflow/models/expandinput.py]: Do we really need this function that’s only used in one place? Especially this one has a different signature from the function on `DictOfListsExpandInput` with the same name.

Comment (hussein-awala) [reply airflow/models/expandinput.py]: Actually we don't

**Review by uranusjr — APPROVED** [review]

**Pull request merged**